parser-typechecker/src/Unison/Lexer.hs

```diff
@@ -854,7 +872,15 @@ lexemes' eof = P.optional space >> do
         inLayout <- S.gets inLayout
         when (not inLayout) $ void $ P.lookAhead (CP.satisfy (/= '}'))
         pure l
+    matchWithBlocks = ["match-with", "cases"]
     parens = open "(" <|> close ["("] (lit ")")
+    brackets = open "[" <|> close ["["] (lit "]")
+    commaSeparator = do
+      env <- S.get
+      case topBlockName (layout env) of
+        Just match | not (match `elem` matchWithBlocks || match == "{") ->
```

Comment:
Suggestion: pull this logic out into a function: `isCommaDelimitedBlock match` and defining it via a positive list of block names, otherwise if I'm reading this right, this guard will become wrong if we ever add another block type that isn't comma-delimited.

Reply:
I think I understand this better... it's really that there is a set of blocks where `,` is not treated specially and it just emits a `Reserved ","`. There are currently only three, `cases`, `match-with`, and `{`.